src/main/java/com/mysema/maven/apt/AbstractProcessorMojo.java

```diff
@@ -422,17 +437,43 @@ public void execute() throws MojoExecutionException {
         File outputDirectory = getOutputDirectory();
         String outputPath = outputDirectory.getAbsolutePath();
         Set<File> directories = new HashSet<File>();        
-        List<String> directoryNames = isForTest() ? project.getTestCompileSourceRoots() 
-                                                  : project.getCompileSourceRoots();
+        List<String> directoryNames = isForTest() ? getTestCompileSourceRoots()
+                                                  : getCompileSourceRoots();
         for (String name : directoryNames) {
             File file = new File(name);
-            if (!file.getAbsolutePath().equals(outputPath) && file.exists()) {
+            if (!file.getAbsolutePath().equals(outputPath) && file.exists() && file.isDirectory()) {
                 directories.add(file);    
             }            
         }
         return directories;
     }
 
+    @SuppressWarnings("unchecked")
+    private List<String> getTestCompileSourceRoots() {
+        if (additionalTestSourceRoots == null) {
+            return project.getTestCompileSourceRoots();
+        }
+        if (getLog().isDebugEnabled()) {
+            getLog().debug("Adding additional test source roots: " + Joiner.on(", ").skipNulls().join(additionalTestSourceRoots));
+        }
+        List<String> sourceRoots = new ArrayList<String>(project.getTestCompileSourceRoots());
+        sourceRoots.addAll(additionalTestSourceRoots);
+        return sourceRoots;
+    }
+
+    @SuppressWarnings("unchecked")
+    private List<String> getCompileSourceRoots() {
+        if (additionalSourceRoots == null) {
+            return project.getCompileSourceRoots();
+        }
+        if (getLog().isDebugEnabled()) {
+            getLog().debug("Adding additional source roots: " + Joiner.on(", ").skipNulls().join(additionalSourceRoots));
+        }
+        List<String> sourceRoots = new ArrayList<String>(project.getCompileSourceRoots());

```

Comment:
How about storing `project.getCompileSourceRoots()` in a temporary variable at the top of the method?
Then you can move the `@SuppressWarnings("unchecked")` to that line.